fix(router): Set route snapshot fragment default value as null

sera-wi opened the pull request:
When a route snapshot is initialized for the first time, this value is undefined. However, when the user refreshes the page, this value is changed to null. This commit sets null as the default value of the fragment.

PR Close #29391

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: 29391


## What is the new behavior?

This commit sets null as the default value of the fragment.

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

sera-wi committed:
fix(router): Set route snapshot fragment default value as null

When a route snapshot is initialized for the first time, this value is undefined. However, when the user refreshes the page, this value is changed to null. This commit sets null as the default value of the fragment.

PR Close #29391

alfaproject commented on packages/router/src/directives/router_link.ts:
```diff
@@ -251,7 +251,7 @@ export class RouterLinkWithHref implements OnChanges, OnDestroy {
    * @see {@link Router#createUrlTree Router#createUrlTree}
    */
   // TODO(issue/24571): remove '!'.
```
I guess you can remove this comment now

sera0731 replied:
I removed the comment. Thank you!

sera-wi committed:
fix(router): Accepted the new golden file

I changed NavigationExtras model's fragment parameter type under goldens/public-api/router since I changed input type of Router to string|null.

atscott commented:
Hi @sera0731 - thanks for the PR. It looks like this would be addressed in #37336. If there are changes here that you would like included there, please add your review to that PR.
